### PR TITLE
Feat: camera reel - photo detail UI

### LINKED ISSUE #2897: Feat: camera reel - photo detail view

### PR DESCRIPTION
**Pull request by lorenzo-ranciaffi**
## What does this PR change?

Fix #2897

Implements the screenshot detail panel for the reels and their metadata

## How to test the changes?

1. Launch the explorer with the CLI arg `camera-reel`, `dclenv zone` and `debug` flags
2. Press "K" to open the reel gallery
3. Click on a photo in order to open the panel

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by popuz — APPROVED** [review]
I left some not so critical comments. Since you already branched from this one, you can apply these changes there (but please don't forgot about them)

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/VisiblePersonController.cs
Comment (lorenzo-ranciaffi) [reply Explorer/Assets/DCL/InWorldCamera/PhotoDetail/VisiblePersonController.cs:94]: as stated in the comment, it is bound to [this](https://github.com/decentraland/unity-explorer/blob/707fd6db601bbcd8f585d7e10ad1b4135fdd9630/Explorer/Assets/DCL/UI/Profile/ProfileElements/ProfileWidgetController.cs#L69) so that as soon as that is solved, this can be solved as well

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/EquippedWearableController.cs
Comment (lorenzo-ranciaffi) [reply Explorer/Assets/DCL/InWorldCamera/PhotoDetail/EquippedWearableController.cs:19]: what do you mean? This is an already existing SO that I needed to use in order to extract some data from it, what do you mean with renaming it?

**Comment by DafGreco** [comment]
Changes verified on Windows ✅ 

Macos currently is not loading the pictures of the gallery so we are currently waiting for a fix in order to approve the PR.

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailPoolManager.cs
Comment (dalkia) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailPoolManager.cs:20]: Is it worth to prewarm this pools?

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailPoolManager.cs
Comment (dalkia) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailPoolManager.cs:104]: Is it worth to reparent? 

- It can bring unnecessary overhead calculations while rebuilding the UI
- You have to maintain extra code for the unused gameobjects.

I would just leave them as child, since that's the only place they can be; unlike other pools

**Review by dalkia — CHANGES_REQUESTED** [review]

File: Explorer/Assets/DCL/Browser/DecentralandUrls/DecentralandUrlsSource.cs
Comment (lorenzo-ranciaffi) [reply Explorer/Assets/DCL/Browser/DecentralandUrls/DecentralandUrlsSource.cs:101]: that's because Dafne was having trouble with the `dclenv` arg to `zone`, so I hardcoded it to make it easier. It will be reverted before merging

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs:93]: Technically, the controller should never know how the view is implemented, whether it uses a button or other control should be transparent. Views should provide just events. In this case, moving the controls to the controller and deleting the view would be the same as it is now. What do you think?

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs:138]: As in the previous comment, controllers should not modify how the view looks by touching its controls directly.

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs:174]: Same as previous comments. The view should provide a method that enables the loading spinner, for example.

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailInfoController.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailInfoController.cs:88]: Same as in previous comments. View should provide events.

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailInfoController.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailInfoController.cs:25]: Please provide a short description about the purpose of the controller and any detail you consider interesting for somebody that finds it for the first time.

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailParameter.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailParameter.cs:7]: Please provide a short description about what's the purpose of the struct, and another description for each public field.

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailInfoView.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailInfoView.cs:7]: Same as in previous comments. If everything is accessable by the controller, then the view loses its purpose. They should be all protected.

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailPoolManager.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailPoolManager.cs:15]: Please provide a short description about what's its purpose.

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailPoolManager.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailPoolManager.cs:96]: Views should provide a method for this, instead of accessing the image directly.

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailUtility.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailUtility.cs:10]: Please provide some documentation.

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailView.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailView.cs:7]: The same as in previous comments about views encapsulation.

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/VisiblePersonController.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/VisiblePersonController.cs:25]: Please provide some short description about the main purpose of the controller.

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/EquippedWearableView.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/PhotoDetail/EquippedWearableView.cs:7]: View should not expose their implementation, just methods and events.

File: Explorer/Assets/DCL/InWorldCamera/ReelActions/ReelCommonActions.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/ReelActions/ReelCommonActions.cs:19]: Plases provide some documentation.

File: Explorer/Assets/DCL/InWorldCamera/ReelActions/ReelCommonActions.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/ReelActions/ReelCommonActions.cs:24]: Plases provide some documentation.

File: Explorer/Assets/DCL/InWorldCamera/ReelActions/ReelUtility.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/ReelActions/ReelUtility.cs:11]: Please provide some documentation.

File: Explorer/Assets/DCL/InWorldCamera/ReelActions/ReelUtility.cs
Comment (QThund) [inline Explorer/Assets/DCL/InWorldCamera/ReelActions/ReelUtility.cs:14]: Please provide some documentation.

File: Explorer/Packages/manifest.json
Comment (QThund) [inline Explorer/Packages/manifest.json:1]: Just in case: Was this on purpose?

**Review by QThund — CHANGES_REQUESTED** [review]
Great work, my comments are mostly related to providing documentation and respecting the MVC pattern.

File: Explorer/Assets/DCL/InWorldCamera/CameraReelGallery/Components/ReelThumbnailController.cs
Comment (lorenzo-ranciaffi) [reply Explorer/Assets/DCL/InWorldCamera/CameraReelGallery/Components/ReelThumbnailController.cs:54]: yes, the intention is to cast the type. As far as I knew multiplying by 1f was "faster" than explicitly casting to float an int. Or are you telling me this to make it more readable?

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs
Comment (lorenzo-ranciaffi) [reply Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs:93]: I agree that a layer of abstraction between the two parts would be much more suited. The main reason why I went with this approach is that I've seen it everywhere in the codebase and doing something different felt misleading from an overall perspective

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs
Comment (lorenzo-ranciaffi) [reply Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs:138]: same as before

File: Explorer/Assets/DCL/InWorldCamera/PhotoDetail/VisiblePersonController.cs
Comment (lorenzo-ranciaffi) [reply Explorer/Assets/DCL/InWorldCamera/PhotoDetail/VisiblePersonController.cs:167]: I guess we could, not sure it's the right PR to do that

File: Explorer/Packages/manifest.json
Comment (lorenzo-ranciaffi) [reply Explorer/Packages/manifest.json:1]: no, it wasn't done on purpose. Probably it is one of those random files that from time to time unity creates and we always avoid to push

File: Explorer/Assets/DCL/InWorldCamera/CameraReelGallery/Components/ReelThumbnailController.cs
Comment (QThund) [reply Explorer/Assets/DCL/InWorldCamera/CameraReelGallery/Components/ReelThumbnailController.cs:54]: I've never heard about that optimization, I will investigate because I like this kind of tricks, although we are talking about nanoseconds. Nevermind.

**Review by QThund — APPROVED** [review]

File: Explorer/Assets/DCL/InWorldCamera/CameraReelGallery/Components/ReelThumbnailController.cs
Comment (lorenzo-ranciaffi) [reply Explorer/Assets/DCL/InWorldCamera/CameraReelGallery/Components/ReelThumbnailController.cs:54]: sure, especially in this case it could probably be considered negligible. Just wanted to make sure that it wasn't a readability thing

**Review by dalkia — APPROVED** [review]

**Pull request merged**
